### PR TITLE
[NUI] Open NUI.ThemeManager and related APIs

### PR DESCRIPTION
**Pull request by rabbitfor**
Signed-off-by: Jiyun Yang <ji.yang@samsung.com>

### API Changes ###

 - ACR: https://code.sec.samsung.net/jira/browse/TCSACR-410

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/View.cs [hunk 1]
```diff
@@ -2589,7 +2586,7 @@ public bool EnableControlStatePropagation
         /// By default, it is false in View, true in Control.
         /// Note that if the value is true, the View will be a touch receptor.
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs:2587]: 말씀하신것이 이론적으로 맞습니다. `EnableControlState`는 기본적인 touch 동작에 아무런 영향을 주어서는 안됩니다. 다만 현재 구조상 그것이 불가능하여 (ControlState를 활성화하기 위해 내부적으로 touch event를 listen 하면, 이것 때문에 z축상 뒤에 존재하는 view에게 touch를 전달할수 없음) touch 동작에 영향을 주고 있기 때문에 그것을 명시하기위해 설명에 써둔것이며, NUI의 touch 구조가 개선되면 해결됩니다.

[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/View.cs:2587]: ControlState를 오픈하면서 함께 오픈해야 맞다고 생각했는데요, 아직 touch 이슈가 해결되지 않은 기능이라 이번 acr에선 제외하도록 하겠습니다.

File: src/Tizen.NUI/src/public/Theme/ThemeManager.cs [hunk 2]
```diff
@@ -48,7 +48,7 @@ static ThemeManager()
         }
 
         /// <summary>
-        /// An event invoked after the theme has changed by <seealso cref="ApplyTheme(Theme)"/>.
+        /// An event invoked after the theme has changed by <see cref="ApplyTheme(Theme)"/>.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/ThemeManager.cs]: Done

File: src/Tizen.NUI/src/public/Theme/ThemeManager.cs [hunk 3]
```diff
@@ -142,7 +142,7 @@ public static ViewStyle GetStyle(string styleName)
         /// </summary>
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/ThemeManager.cs:143]: Done

File: src/Tizen.NUI/src/public/Theme/Theme.cs [hunk 4]
```diff
@@ -293,9 +305,9 @@ public void Merge(string xamlFile)
             MergeWithoutClone(new Theme(xamlFile));
         }
 
-        /// <summary>Merge other Theme into this.</summary>
-        /// <param name="theme">The Theme.</param>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <summary>Merge other theme into this.</summary>
+        /// <param name="theme">The theme to be merge into this theme.</param>
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/Theme.cs]: Done

**Review by hyunjushin — APPROVED** [review]

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 5]
```diff
@@ -122,48 +128,67 @@ public Position2D Position2D
...
         public Position PivotPoint
         {
             get => (Position)GetValue(PivotPointProperty);
             set => SetValue(PivotPointProperty, value);
         }
 
-        /// This will be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <summary>
+        /// Gets or sets the size width of the view.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:169]: could probably drop "size" -> "the width of the view".  same with the next one

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 6]
```diff
@@ -274,25 +313,32 @@ public Size2D MaximumSize
             set => SetValue(ClippingModeProperty, value);
         }
 
-        /// This will be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <summary>
+        /// Sets the size of a view for the width, the height and the depth.<br />
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:317]: consider oxford comma: "the width, the height, and the depth"

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 7]
```diff
@@ -274,25 +313,32 @@ public Size2D MaximumSize
...
+        /// The views default depth is the minimum of width and height.<br />
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:318]: when is the default used? Size object always has depth assigned. Is the default really min(width, height)?

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 8]
```diff
@@ -274,25 +313,32 @@ public Size2D MaximumSize
...
             get => (Extents)GetValue(MarginProperty) ?? (margin = new Extents());
             set => SetValue(MarginProperty, value);
         }
 
-        /// <summary> View BackgroundColor </summary>
-        /// This will be public opened in tizen_5.5 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <summary>
+        /// Gets or sets the color of the background of view.
+        /// The mutually exclusive with "backgroundImage".
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:339]: what happens if both are set? is it a fatal error or one is ignored? which one? Does setting it overwrites backgroundImage?

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 9]
```diff
@@ -485,7 +530,11 @@ public override void CopyFrom(BindableObject other)
         /// <summary>
         /// Release instance.
         /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <param name="disposing">
+        /// If disposing equals true, the method has been called directly or indirectly by a user's code. Managed and unmanaged resources can be disposed.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:534]: It's nice to have this detailed description, but this is a standard method and the description could have been simpler.

File: src/Tizen.NUI/src/public/Theme/Theme.cs [hunk 10]
```diff
@@ -89,13 +93,13 @@ public Theme(string xamlFile) : this()
         /// <summary>
         /// The string key to identify the Theme.
         /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <since_tizen> 9 </since_tizen>
         public string Id { get; set; }
 
         /// <summary>
         /// The string key to identify the Theme.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Theme/Theme.cs:100]: repeated summary, see line 94

File: src/Tizen.NUI/src/public/Theme/Theme.cs [hunk 11]
```diff
@@ -224,21 +228,22 @@ internal ResourceDictionary Resources
         /// Gets a style of given style name.
         /// </summary>
         /// <param name="styleName">The string key to find a ViewStyle.</param>
-        /// <returns>Founded style instance.</returns>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <returns>Founded style instance if the style name is found, otherwise null.</returns>
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Theme/Theme.cs:231]: Founded -> Found,   name is found -> name has been found

Please replace other ocurrences of "founded" as well

File: src/Tizen.NUI/src/public/Theme/ThemeChangedEventArgs.cs [hunk 12]
```diff
@@ -33,8 +34,9 @@ public ThemeChangedEventArgs(string themeId)
         }
 
         /// <summary>
+        /// The new theme's id.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Theme/ThemeChangedEventArgs.cs:37]: usually in the text id -> ID

File: src/Tizen.NUI/src/public/Theme/ThemeManager.cs [hunk 13]
```diff
@@ -126,23 +126,24 @@ public static bool ApplyBaseTheme(string themeId)
 
         /// <summary>
         /// Load a style with style name in the current theme.
-        /// For components, the style name is a component name (e.g. Button) in normal case.
+        /// For components, the style name is a component name with namespace (e.g. Tizen.NUI.Components.Button) in normal case.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Theme/ThemeManager.cs:129]: by "in normal case" you mean that it is usually that name, or by convention that name, not the lower/upper case, right?

File: src/Tizen.NUI/src/public/BaseComponents/ControlState.cs [hunk 14]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright(c) 2020 Samsung Electronics Co., Ltd.
+ * Copyright(c) 2021 Samsung Electronics Co., Ltd.
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/ControlState.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/ControlState.cs [hunk 15]
```diff
@@ -23,43 +23,44 @@
...
-    [EditorBrowsable(EditorBrowsableState.Never)]
+    /// <since_tizen> 9 </since_tizen>
     [Binding.TypeConverter(typeof(ControlStateTypeConverter))]
     public class ControlState : IEquatable<ControlState>
     {
         private static readonly Dictionary<string, ControlState> stateDictionary = new Dictionary<string, ControlState>();
         //Default States
         /// <summary>
-        /// All State.
+        /// Thia is used in a selector class. It represents all states, so if this state is defined in a selector, the other states are ignored.
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/ControlState.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs [hunk 16]
```diff
@@ -23,16 +23,14 @@
 namespace Tizen.NUI.BaseComponents
 {
     /// <summary>
-    /// Selector class, which is related by Control State, it is base class for other Selector.
+    /// The selector class is a collection of a <see cref="ControlState" /> and a T value pair.
     /// </summary>
-    /// <typeparam name="T">The property type of the selector. if it's reference type, it should be of type <see cref="ICloneable"/> that implement deep copy in <see cref="ICloneable.Clone"/>.</typeparam>
-    /// <since_tizen> 6 </since_tizen>
-    /// This will be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API.
+    /// <typeparam name="T">The property type of the selector. If it's reference type, it is recommended to be of type <see cref="ICloneable"/>.</typeparam>
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs (hunk 5 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs (hunk 6 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs (hunk 7 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs]: Done. I removed the line explaining default depth.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs (hunk 8 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs]: Setting it overwrites backgroundImage. I added more description. Please check.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs (hunk 9 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs]: Done. I made it simpler.

File: src/Tizen.NUI/src/public/Theme/Theme.cs (hunk 11 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/Theme.cs]: Done.

File: src/Tizen.NUI/src/public/Theme/ThemeChangedEventArgs.cs (hunk 12 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/ThemeChangedEventArgs.cs]: Done.

File: src/Tizen.NUI/src/public/Theme/ThemeManager.cs (hunk 13 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/ThemeManager.cs]: I edited the description. Please check.

File: src/Tizen.NUI/src/public/BaseComponents/ControlState.cs [hunk 17]
```diff
@@ -82,9 +83,9 @@ public class ControlState : IEquatable<ControlState>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public static readonly ControlState SelectedFocused = Selected + Focused;
         /// <summary>
-        /// Other State.
+        /// This is used in a selector class. It represents all other states except for states that are already defined in a selector.
```
[the remark this answers is not in the document]
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/BaseComponents/ControlState.cs:86]: For example,
if a selector defines 3 states `Selected`, `Pressed` and `Other` and a control is a `Disabled` state which is not defined in the selector, the control gets `Other` value in the selector.

File: src/Tizen.NUI/src/public/Theme/Theme.cs (hunk 10 above)
Comment (rabbitfor) [reply src/Tizen.NUI/src/public/Theme/Theme.cs]: Done.

File: src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs [hunk 18]
```diff
@@ -72,8 +73,11 @@ public ViewStyle(ViewStyle viewAttributes)
             CopyFrom(viewAttributes);
         }
 
-        /// This will be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API.
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <summary>
+        /// Gets or sets the image resource url of the background of view.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/Style/ViewStyle.cs:77]: url -> URL, abbreviations in text should be uppercase

**Pull request merged**